cinder/volume/api.py

```diff
@@ -1379,6 +1382,14 @@ def _roll_back_status():
                                                     'size': volume.size})
             raise exception.InvalidInput(reason=msg)
 
+        # Make sure we pass the potential size limitations in the volume type
+        try:
+            volume_type = volume_types.get_volume_type(context,
+                                                       volume.volume_type_id)
+        except (exception.InvalidVolumeType, exception.VolumeTypeNotFound):
+            volume_type = None
```

Comment:
How can this happen? Which volumes don't hav a valid volume-type? Is `volume_type_id` just empty and we would thus use the default?

Reply:
The volume type id can be none here for volumes that were created without a default volume type.